Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -202,9 +202,9 @@ struct Gx2FitterResult {
   Result<void> result{Result<void>::success()};
 
   // collectors
-  std::vector<ActsVector<2>> collectorResiduals;
-  std::vector<ActsSquareMatrix<2>> collectorCovariance;
-  std::vector<BoundMatrix> collectorJacobians;
+  std::vector<ActsScalar> collectorResiduals;
+  std::vector<ActsScalar> collectorCovariances;
```

Comment:
Where do you store the information about the dimensionality of the measurement here?

Reply:
I don't store it at all. Each n-dimensional measurement gets split up into n 1-dimensional measurements. This is allowed since all dimensions are (so far) independent of each other.